CHEWBBACA/CHEWBBACA_NS/sync_schema.py

```diff
@@ -0,0 +1,1330 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+Purpose
+-------
+
+This module allows users to synchronize a local schema, previously
+downloaded from the Chewie-NS, with the remote schema in the Chewie-NS.
+
+The synchronization process will retrieve new alleles added to the remote
+schema after the last synchronization date and update the local schema,
+ensuring that local and remote alleles have the same integer identifiers.
+The process also allows users to submit novel alleles and update the remote
+schema. Novel alleles that are not submitted keep a '*' in the identifier.
+
+Expected input
+--------------
+
+The process expects the following variables wheter through command line
+execution or invocation of the :py:func:`main` function:
+
+- ``-sc``, ``schema_directory`` : Path to the directory with the schema to be
+  synced.
+
+    - e.g.: ``/home/user/chewie_schemas/ecoli_schema``
+
+- ``--cpu``, ``cpu_cores`` : Number of CPU cores that will be used to
+  determine new representatives if the process downloads new alleles from
+  the Chewie-NS.
+
+    - e.g.: ``4``
+
+- ``--ns_url``, ``nomenclature_server_url`` : The base URL for the
+  Nomenclature Server.
+
+    - e.g.: ``http://127.0.0.1:5000/NS/api/`` (localhost)
+
+- ``--submit`` : If the process should identify new alleles in the local
+  schema and send them to the Chewie-NS (only registered users can submit
+  new alleles).
+
+Code documentation
+------------------
+"""
+
+
+import os
+import sys
+import json
+import shutil
+import pickle
+import hashlib
+import argparse
+import requests
+import datetime as dt
+from copy import deepcopy
+from collections import defaultdict
+from SPARQLWrapper import SPARQLWrapper
+from urllib3.exceptions import InsecureRequestWarning
+
+from Bio import SeqIO
+
+try:
+    from utils import constants as cnst
+    from utils import auxiliary_functions as aux
+    from PrepExternalSchema import PrepExternalSchema
+except:
+    from CHEWBBACA.utils import constants as cnst
+    from CHEWBBACA.utils import auxiliary_functions as aux
+    from CHEWBBACA.PrepExternalSchema import PrepExternalSchema
+
+
+# Suppress only the single warning from urllib3 needed.
+requests.packages.urllib3.disable_warnings(category=InsecureRequestWarning)
```

Comment:
The `InsecureRequestWarning` suppression can commented out because the warning will only be raised in local Chewie-NS instances with self-signed certificates. The same can be applied to the the warning import.